### ISSUE TITLE
Scripted Crowdsales AKA CrowdScript

### ISSUE THREAD

**Issue by ripper234** [OmniLayer/spec#214]
We current have one type of crowdsale with one set of rules, which is pretty identical to the original Mastercoin crowdsale of August 2013.

We also have some ideas on how to expand this:
- [Capped Crowdsale](https://github.com/mastercoin-MSC/spec/issues/205)
- [Cascading Price Crowdsale](https://github.com/mastercoin-MSC/spec/issues/201)
- Daily sale
- What I'll dub "Order based discount crowdsale" - giving the early bird bonus to the first tokens purchased, instead of by time (e.g. First purchased token costs X, each new token costs Y*1.001 where Y is the last sold token).

We'll probablly have more and more ideas over time.

Instead of trying to implement all of these, how about we borrow some ideas from Ethereum and Bitcoin Scripting and implement a very limited scripting language to handle the math of the crowdsale?

Imagine the issuer of the crowdsale being allowed to pick from a few pre-defined options, but also being able to submit a custom script in this language (I'll dub that CrowdScript as a temporary name). Here is a [sample CrowdScript](https://gist.github.com/ripper234/aafec73653deee9b8cd0) that implements the original Mastercoin crowdsale. Implementing the other order types using the above primitives is left as an exercise for the reader (if you do this exercise please post here!).

To recap - I suggest we build this scripting language bit by bit, INSTEAD of trying to implement hardcoded cases. I think it's more valuable to us long term, and building an MVP scripting language won't take more than 2-4 weeks (if you have other estimates, feel free to share them!).
